lib/src/main/java/growthbook/sdk/java/ExperimentEvaluator.java

```diff
@@ -1,36 +1,35 @@
 package growthbook.sdk.java;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
 import javax.annotation.Nullable;
 import com.google.gson.JsonObject;
+import growthbook.sdk.java.multiusermode.configurations.EvaluationContext;
+import growthbook.sdk.java.multiusermode.configurations.UserContext;
+import growthbook.sdk.java.multiusermode.usage.TrackingCallbackWithUser;
 import lombok.extern.slf4j.Slf4j;
 
 /**
  * <b>INTERNAL</b>: Implementation of experiment evaluation
  */
 @Slf4j
-class ExperimentEvaluator implements IExperimentEvaluator {
+public class ExperimentEvaluator implements IExperimentEvaluator {
 
     private final ConditionEvaluator conditionEvaluator = new ConditionEvaluator();
     private final GrowthBookJsonUtils jsonUtils = GrowthBookJsonUtils.getInstance();
+    private final Set<String> trackedExperiments = new HashSet<>();
```

Comment:
I'm a little worried about having this here.  With the MultiUser mode, this set will continue to grow unbounded.  Millions of users = millions of entries in the set.

In JavaScript, I solved this by removing this logic completely and leaving it up to the user to de-duplicate tracking calls.  Another option is to turn this into an LRU cache that only keeps the most recent X entries.

Note: This was not an issue before since we were persisting this in the GrowthBook instance, which was for a single user and short lived.

Reply:
I'm happy to convert that into an LRU like solution. I wasn't completely clear about the actual need behind this though! Today, in the current implementation adds all/any experiment to this set and doesn't clear it.

Reply:
But in the current implementation, the GrowthBook instances are short-lived (one per-request), so the set will not grow unbounded.  We're changing that with the multi-user class, so will need to implement our own method of keeping the set from growing too large.